lib/rubocop/cop/layout/trailing_whitespace.rb

```diff
@@ -123,7 +123,11 @@ def extract_heredocs(ast)
         end
 
         def offense_range(lineno, line)
-          source_range(processed_source.buffer, lineno, (line.rstrip.length)...(line.length))
+          source_range(
+            processed_source.buffer,
+            lineno,
+            (line.sub(/[[:blank:]]+\z/, '').length)...(line.length)
```

Comment:
My two cents. It seems that it can be written in a single line.

```suggestion
            processed_source.buffer, lineno, (line.sub(/[[:blank:]]+\z/, '').length)...(line.length)
```

Reply:
Ah sorry, I missed it. Done.